lib/models/location/location.dart

```diff
@@ -12,4 +12,18 @@ class Location with _$Location {
 
   factory Location.fromJson(Map<String, Object?> json) =>
       _$LocationFromJson(json);
+
+  static isValidLocation(Location? location) {
+    if (location == null) return false;
+    if (location.latitude == null || location.longitude == null) return false;
+    final latValue = location.latitude!;
+    final longValue = location.longitude!;
+    if (latValue.isNaN || latValue.isInfinite || latValue == 0.0) {
+      return false;
+    }
+    if (longValue.isNaN || longValue.isInfinite || longValue == 0.0) {
```

Comment:
Latitude and Longitude of 0 are valid values.  The Prime Meridian is even a tourist attraction 😄 something like NaN should be used to store invalid values.

Reply:
Hey, thanks for pointing this out! Have merged a PR to fix this. Now, a location will be considered invalid only if both latitude and longitude are 0. We know that (0,0) is a valid location, but some older Android versions were causing trouble by tagging location as (0,0). We put in a fix and a migration to handle this. For the time being, we're gonna keep treating (0,0) as invalid to avoid any issues.

Reply:
Ah it's the worst when technically valid values like 0,0 are used as invalid.  Thanks for fixing!  👍